pallets/msa/src/lib.rs

```diff
@@ -329,6 +332,12 @@ pub mod pallet {
 				Error::<T>::UnauthorizedProvider
 			);
 
+			// Verify that the provider is a registered provider
+			ensure!(
+				!Self::get_provider_metadata(Provider(provider_msa_id)).is_none(),
```

Comment:
Do we want to use `is_registered_provider` here?

Reply:
Yes.  I just pushed the commit with it.